### DISCUSSION TITLE
Editing self-edited objects

### DISCUSSION THREAD

**Discussion by OpenGreenStreet** [streetcomplete/StreetComplete#2407] [Ideas]
I would like to be able to edit self-edited objects at any later time.
(For example, I could update changed opening hours in StreetComplete at a later time. Each final edited object could get a "green point" instead of "no point").

(de: Editieren von selbst bearbeiteten Objekten: Ich möchte, selbst bearbeiteten Objekten zu einem beliebigen späteren Zeitpunkt bearbeiten können. (Zum Beispiel könnte ich so geänderte Öffnungszeiten in StreetComplete zu einem späteren Zeitpunkt aktualisieren. Jedes abschließend bearbeitete Objekt, könnte dazu statt "keinem Punkt" einen „grünen Punkt“ erhalten.)

**Comment by smichel17** [comment]
Recently, resurvey quests were added. So, after some amount of time since the quest was last answered, StreetComplete will ask if the map data is still up to date. For opening hours, I think it is asked every year, and there is a setting in the app for asking more often (every 6 months). It is not "at any time", but maybe it is good enough?

**Comment by OpenGreenStreet** [comment]
Thanks for the tip. But "anytime" and all object types would be nicer ...

**Reply by smichel17** [reply]
Well, I think there are resurvey quests for all quests that are asked. So "all [supported] object types" is already there.

For "any time" — it is not a goal of streetcomplete to be a general purpose editor, like Vespucci, but more of a tool for casual users or people new to OSM. So, there is currently no way to directly select a node on the map — only answer quests for it — and I do not think @westnordost would want to add it.

I could imagine some way to mark that the information is outdated, which would cause the resurvey quests to show up again immediately. But only for certain types of quests, since SC does not show information like opening hours on the map, only in the quest bubble. So, this is probably too much work to implement for too little gain; in this case it would be better to use a different app.

I could also imagine a setting that would show all resurvey quests, regardless of how recently they were surveyed. This would be a more reasonable amount of work to implement, but still requires adding an easier way to switch which quest is shown, when there are multiple quests for one object, otherwise you would have to repeatedly hide quests for the object you are re-surveying until you reached the quest that you wanted to answer. I don't think this would be worth implementing for this feature alone, but it is also useful just for a regular user answering quests, so it is a better use of time.

**Reply by OpenGreenStreet** [reply]
How exactly, is not so important for me. But that I can trigger a  resurvey quests would be useful. (Maybe via long tap on an object...)

**Reply by westnordost** [reply]
Hmm, you could set the time and date of your smartphone to a few years in the future ;-)

But anyway, there is also a setting for it: You can control to be asked less often (:2) or more often (x2) about resurveying things

**Reply by matkoniecz** [reply]
"Hmm, you could set the time and date of your smartphone to a few years in the future ;-)" - just in case someone will try this - it will result in adding nonsense dates in some tags.

**Reply by westnordost** [reply]
Oh, right! Yeah, so you shouldn't do this. It would tag that you checked this and that 4 years in the future.

**Comment by westnordost** [comment]
What you propose will likely never be part of StreetComplete, this is out of scope. However, you might want to try out this app: https://play.google.com/store/apps/details?id=fr.dogeo.osmgo

**Reply by westnordost** [reply]
Or, more powerful and mature: https://play.google.com/store/apps/details?id=de.blau.android
